packages/server/lib/controllers/sync.controller.ts

```diff
@@ -118,14 +118,42 @@ class SyncController {
                 return;
             }
 
-            const syncs = await getSyncs(connection, orchestrator);
+            const rawSyncs = await getSyncs(connection, orchestrator);
+
+            if (!connection.id) {
+                res.send(rawSyncs);
+                return;
+            }
+
+            const syncs = await this.addObjectCount(rawSyncs, connection.id);
 
             res.send(syncs);
         } catch (e) {
             next(e);
         }
     }
 
+    private async addObjectCount(syncs: (Sync & { models: string[] })[], connectionId: number) {
+        const models = syncs.map((sync) => sync.models).flat();
+        const objectCountByModel: Record<string, number> = {};
+        for (const model of models) {
+            if (!(model in objectCountByModel)) {
+                const count = await recordsService.getRecordsCount({ connectionId, model });
```

Comment:
ideally here you select all count of envId + connectionId and then find the appropriate Model count in the for loop so we can do it a one query instead of N

Reply:
Good suggestion - much cleaner approach